dmoj/generator.py

```diff
@@ -6,20 +6,16 @@
 
 
 class GeneratorManager(object):
-    def __init__(self):
-        self._cache = {}
-
-    def get_generator(self, filename, flags):
+    def get_generator(self, filenames, flags):
         from dmoj.executors import executors
 
-        filename = os.path.abspath(filename)
-        cache_key = filename, tuple(flags)
-        if cache_key in self._cache:
-            return self._cache[cache_key]
+        filenames = map(os.path.abspath, filenames)
+        sources = {}
 
         try:
-            with open(filename) as file:
-                source = file.read()
+            for filename in filenames:
+                with open(filename, 'r') as f:
```

Comment:
From an instance in prod:

Given this `generator` node in the `init.yml`
```
generator:
    source: generator.cpp
    time_limit: 20
    memory_limit: 262144
```

We obtain the following traceback:
```
Traceback (most recent call last):
    File "<redacted>/judge/dmoj/generator.py", line 17, in get_generator
       with open(filename, 'r') as f:
IOError: [Errno 2] No such file or directory: u'<redacted>/problems/peths/ds4/time_limit'
```